views/default/object/elements/summary.php

```diff
@@ -44,15 +44,12 @@
 
 $icon = elgg_extract('icon', $vars);
 if (isset($icon)) {
-	$params = $vars;
-	foreach (['title', 'metadata', 'subtitle', 'tags', 'content', 'icon', 'class'] as $key) {
-		unset($params[$key]);
-	}
 	$class = (array) elgg_extract('class', $vars, []);
 	$class[] = 'elgg-listing-summary';
-	$params['class'] = $class;
-
-	$params['data-guid'] = $entity->guid;
+	$params = [
```

Comment:
I think it's fine for now. I don't think we will be passing image block vars from the elgg_list_entities(). In 3.0, we can just wrap it in a proper view.